### PR TITLE
Make kernel parse-tree search case insensitive

### LINKED ISSUE #345: Kernel type match is case sensitive
xed67771@arcgdel01:~/proj/lfric_trunk$ psyclone gungho/source/algorithm/runtime_constants_mod.x90 -api dynamo0.3 -oalg alg.f90 -opsy psy.f90 -d gungho/source/kernel
'Parse Error: Kernel type calc_dA_at_w2_kernel_type does not exist'

The problem is the case sensitive match in file kernel.py function get_kernel_metadata

### PR DESCRIPTION
**Pull request by pelson**
Closes #345

### CODE REVIEW

File: src/psyclone/parse/kernel.py [hunk 1]
```diff
@@ -601,8 +601,9 @@ def get_kernel_metadata(name, ast):
     ktype = None
     for statement, _ in fpapi.walk(ast, -1):
         if isinstance(statement, fparser1.block_statements.Type) \
-           and statement.name == name:
+           and statement.name.lower() == name.lower():
             ktype = statement
+            break
```
Comment (pelson) [inline src/psyclone/parse/kernel.py]: A bit of a micro-optimisation.

File: src/psyclone/parse/kernel.py [hunk 2]
```diff
@@ -586,7 +586,7 @@ def get_kernel_metadata(name, ast):
     of the parse tree (a Fortran type) with the name 'name'.
 
     :param str name: the metadata name (of a Fortran type). Also \
-    the name referencing the kernel in the algorithm layer)
+    the name referencing the kernel in the algorithm layer). Case insensitive.
```
Comment (arporter) [inline src/psyclone/parse/kernel.py]: Not your mistake originally but please could you remove the ")" after "algorithm layer".

File: src/psyclone/tests/parse/kernel_test.py [hunk 3]
```diff
@@ -131,6 +131,31 @@ def test_getkernelfilepath_caseinsensitive2(tmpdir):
     assert "tmp" in result
     assert "test_mod.f90" in result
 
+# function get_kernel_metadata
+
+def test_get_kernel_metadata_no_match():
```
Comment (arporter) [inline src/psyclone/tests/parse/kernel_test.py]: Thanks for adding these tests. Please could you add docstrings for them.

**Comment by arporter** [comment]
Thanks for doing the docstrings. All good now. I'll just bring the branch up-to-date with master and wait to see that Travis is happy.

**Pull request merged**